File: src/main/java/kirjanpito/ui/javafx/dialogs/SettingsDialogFX.java
package kirjanpito.ui.javafx.dialogs;

import javafx.geometry.Insets;
import javafx.geometry.Pos;
import javafx.scene.Scene;
import javafx.scene.control.*;
import javafx.scene.layout.*;
import javafx.stage.Modality;
import javafx.stage.Stage;
import javafx.stage.Window;

import kirjanpito.util.AppSettings;

/**
 * JavaFX yleiset asetukset -dialogi.
 */
public class SettingsDialogFX {
    
    private Stage dialog;
    private CheckBox autoSaveCheck;
    private CheckBox showVatColumnCheck;
    private Spinner<Integer> backupIntervalSpinner;
    
    private AppSettings settings;
    
    public SettingsDialogFX(Window owner) {
        settings = AppSettings.getInstance();
        
        dialog = new Stage();
        dialog.initModality(Modality.APPLICATION_MODAL);
        dialog.initOwner(owner);
        dialog.setTitle("Yleiset asetukset");
        dialog.setResizable(false);
        
        createContent();
    }
    
    private void createContent() {
        VBox root = new VBox(20);
        root.setPadding(new Insets(20));
        root.setStyle("-fx-background-color: #ffffff;");
        
        // General settings
        TitledPane generalPane = new TitledPane();
        generalPane.setText("Kirjaus");
        generalPane.setCollapsible(false);
        
        VBox generalBox = new VBox(10);
        generalBox.setPadding(new Insets(10));
        
        autoSaveCheck = new CheckBox("Tallenna automaattisesti siirryttäessä tositteesta toiseen");
        autoSaveCheck.setSelected(settings.getBoolean("autoSave", true));
        
        showVatColumnCheck = new CheckBox("Näytä ALV-sarake vientitaulukossa");
        showVatColumnCheck.setSelected(settings.getBoolean("showVatColumn", true));
        
        generalBox.getChildren().addAll(autoSaveCheck, showVatColumnCheck);
        generalPane.setContent(generalBox);
        
        // Backup settings
        TitledPane backupPane = new TitledPane();
        backupPane.setText("Varmuuskopiointi");
        backupPane.setCollapsible(false);
        
        VBox backupBox = new VBox(10);
        backupBox.setPadding(new Insets(10));
        
        HBox intervalBox = new HBox(10);
        intervalBox.setAlignment(Pos.CENTER_LEFT);
        Label intervalLabel = new Label("Varmuuskopiointiväli (minuuttia):");
        backupIntervalSpinner = new Spinner<>(0, 60, settings.getInt("backupInterval", 10));
        backupIntervalSpinner.setEditable(true);
        backupIntervalSpinner.setPrefWidth(80);
        intervalBox.getChildren().addAll(intervalLabel, backupIntervalSpinner);
        
        backupBox.getChildren().add(intervalBox);
        backupPane.setContent(backupBox);
        
        // Buttons
        HBox buttonBox = new HBox(10);
        buttonBox.setAlignment(Pos.CENTER_RIGHT);
        
        Button saveBtn = new Button("Tallenna");
        saveBtn.setDefaultButton(true);
        saveBtn.setStyle("-fx-background-color: #2563eb; -fx-text-fill: white;");
        saveBtn.setOnAction(e -> save());
        
        Button cancelBtn = new Button("Peruuta");
        cancelBtn.setCancelButton(true);
        cancelBtn.setOnAction(e -> dialog.close());
        
        buttonBox.getChildren().addAll(cancelBtn, saveBtn);
        
        root.getChildren().addAll(generalPane, backupPane, buttonBox);
        
        Scene scene = new Scene(root, 450, 280);
        dialog.setScene(scene);
    }
    
    private void save() {
        settings.set("autoSave", autoSaveCheck.isSelected());
        settings.set("showVatColumn", showVatColumnCheck.isSelected());
        settings.set("backupInterval", backupIntervalSpinner.getValue());
        
        settings.save();
        dialog.close();
    }
    
    public void show() {
        dialog.showAndWait();
    }
}

File: src/main/kotlin/kirjanpito/ui/javafx/dialogs/SettingsDialogFX.kt
package kirjanpito.ui.javafx.dialogs

import javafx.geometry.Insets
import javafx.geometry.Pos
import javafx.scene.Scene
import javafx.scene.control.*
import javafx.scene.layout.HBox
import javafx.scene.layout.VBox
import javafx.stage.Modality
import javafx.stage.Stage
import javafx.stage.Window
import kirjanpito.util.AppSettings

/**
 * JavaFX yleiset asetukset -dialogi.
 */
class SettingsDialogFX(owner: Window?) {
    
    private val dialog: Stage = Stage()
    private lateinit var autoSaveCheck: CheckBox
    private lateinit var showVatColumnCheck: CheckBox
    private lateinit var backupIntervalSpinner: Spinner<Int>
    
    private val settings: AppSettings = AppSettings.getInstance()
    
    init {
        dialog.initModality(Modality.APPLICATION_MODAL)
        dialog.initOwner(owner)
        dialog.title = "Yleiset asetukset"
        dialog.isResizable = false
        
        createContent()
    }
    
    private fun createContent() {
        val root = VBox(20.0).apply {
            padding = Insets(20.0)
            style = "-fx-background-color: #ffffff;"
        }
        
        // General settings
        val generalPane = TitledPane().apply {
            text = "Kirjaus"
            isCollapsible = false
        }
        
        val generalBox = VBox(10.0).apply {
            padding = Insets(10.0)
        }
        
        autoSaveCheck = CheckBox("Tallenna automaattisesti siirryttäessä tositteesta toiseen").apply {
            isSelected = settings.getBoolean("autoSave", true)
        }
        
        showVatColumnCheck = CheckBox("Näytä ALV-sarake vientitaulukossa").apply {
            isSelected = settings.getBoolean("showVatColumn", true)
        }
        
        generalBox.children.addAll(autoSaveCheck, showVatColumnCheck)
        generalPane.content = generalBox
        
        // Backup settings
        val backupPane = TitledPane().apply {
            text = "Varmuuskopiointi"
            isCollapsible = false
        }
        
        val backupBox = VBox(10.0).apply {
            padding = Insets(10.0)
        }
        
        val intervalBox = HBox(10.0).apply {
            alignment = Pos.CENTER_LEFT
        }
        
        val intervalLabel = Label("Varmuuskopiointiväli (minuuttia):")
        
        backupIntervalSpinner = Spinner<Int>(0, 60, settings.getInt("backupInterval", 10)).apply {
            isEditable = true
            prefWidth = 80.0
        }
        
        intervalBox.children.addAll(intervalLabel, backupIntervalSpinner)
        backupBox.children.add(intervalBox)
        backupPane.content = backupBox
        
        // Buttons
        val buttonBox = HBox(10.0).apply {
            alignment = Pos.CENTER_RIGHT
        }
        
        val saveBtn = Button("Tallenna").apply {
            isDefaultButton = true
            style = "-fx-background-color: #2563eb; -fx-text-fill: white;"
            setOnAction { save() }
        }
        
        val cancelBtn = Button("Peruuta").apply {
            isCancelButton = true
            setOnAction { dialog.close() }
        }
        
        buttonBox.children.addAll(cancelBtn, saveBtn)
        
        root.children.addAll(generalPane, backupPane, buttonBox)
        
        val scene = Scene(root, 450.0, 280.0)
        dialog.scene = scene
    }
    
    private fun save() {
        settings.set("autoSave", autoSaveCheck.isSelected)
        settings.set("showVatColumn", showVatColumnCheck.isSelected)
        settings.set("backupInterval", backupIntervalSpinner.value)
        
        settings.save()
        dialog.close()
    }
    
    fun show() {
        dialog.showAndWait()
    }
}
